R/zchunk_L203.water.mapping.R

```diff
@@ -38,103 +37,134 @@ module_water_L203.water.mapping_DISABLED <- function(command, ...) {
+      # ^^ join GLU names, which will replace GLU codes in this tibble
+      select(-GLU) %>% rename(GLU = GLU_name) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      repeat_add_columns(filter(A03.sector, water.sector %in% IRRIGATION)) %>%
+      repeat_add_columns(tibble(water_type = c("water consumption", "water withdrawals"))) %>%
+      mutate(supplysector = set_water_input_name(water.sector, water_type, A03.sector, GLU)) ->
+      L203.mapping_irr
+
+    # (b) non-irrigation sectors
+    GCAM_region_names %>%
+      repeat_add_columns(filter(A03.sector, !(water.sector %in% IRRIGATION))) %>%
+      mutate(GLU = NA) %>%
+      repeat_add_columns(tibble(water_type = c("water consumption", "water withdrawals"))) %>%
+      mutate(supplysector = set_water_input_name(water.sector, water_type, A03.sector)) ->
+      L203.mapping_nonirr
+
+    # (c) combine irrigation and non-irrigation sectors and add additional required columns
+    L203.mapping_irr %>%
+      bind_rows(L203.mapping_nonirr) %>%
+      mutate(coefficient = 1,
+             subsector = supplysector,
+             technology = supplysector,
+             logit.year.fillout = first(BASE_YEARS)) %>%
+      arrange(GCAM_region_ID) %>%
+      left_join(select(L165.ag_IrrEff_R, -field.eff), by = "GCAM_region_ID") %>%
+      # ^^ non-restrictive join required (NA values generated for region 30, Taiwan)
+      mutate(coefficient = if_else(water.sector == IRRIGATION & water_type == "water withdrawals",
+                                   1 / conveyance.eff, coefficient)) %>%
+      # ^^ conveyance losses for irrigation--applied to withdrawals only
+      select(-conveyance.eff) ->
+      L203.mapping_all
+
+    # Sector information
+    L203.mapping_all %>%
+      select(one_of(c(LEVEL2_DATA_NAMES[["Supplysector"]], "logit.type"))) ->
+      L203.Supplysector
+
+    # Subsector logit exponents for mapping sector
+    L203.mapping_all %>%
+      select(one_of(c(LEVEL2_DATA_NAMES[["SubsectorLogit"]], "logit.type"))) ->
+      L203.SubsectorLogit
+
+    # Subsector share weights to 1 (no competition)
+    L203.mapping_all %>%
+      mutate(share.weight = 1,
+             year.fillout = first(MODEL_YEARS)) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["SubsectorShrwtFllt"]])) ->
+      L203.SubsectorShrwtFllt
+
+    # Pass-through technology to the water resource (no competition)
+    L203.mapping_all %>%
+      repeat_add_columns(tibble(year = MODEL_YEARS)) %>%
+      mutate(share.weight = 1) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["TechShrwt"]])) ->
+      L203.TechShrwt
+
+    # Pass-through technology to the water resource
+    L203.mapping_all %>%
+      repeat_add_columns(tibble(year = MODEL_YEARS)) %>%
+      mutate(minicam.energy.input = water_type,
+             market.name = region) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["TechCoef"]])) ->
+      L203.TechCoef
+
+
+    # OUTPUTS
+    L203.Supplysector %>%
```

Comment:
I would consider creating the water mapping files by demand types (irr, dom, ind, ...) rather than by variable type and break up the final water mapping xml file by demand types to reduce the file size and allow ease of modification. But this is for future work too.